### PR TITLE
Add delete propagation for requestType

### PR DESCRIPTION
**Pull request by kevin-wangg**
Not sure if this is the proper graphql way to do this (ie. can you have mutations call each other?)

### CODE REVIEW

**Review by meganniu — APPROVED** [review]
LGTM!!

**Review by guy-ro — APPROVED** [review]
LGTM

don't forget to update this for the new requestType schema

**Review by Anand-Issac — APPROVED** [review]
LGTM

File: server/graphql/resolvers.ts [hunk 1]
```diff
@@ -5,6 +5,12 @@ import { RequestTypeInterface } from '../models/requestTypeModel'
 import { ServerResponseInterface } from './serverResponse'
 import { Types } from 'mongoose'
 
+const softDeleteRequestTypeHelper = (id, dataSources): Promise<ServerResponseInterface> => {
+  const requestType = dataSources.requestTypes.getById(id)
+  requestType.requests.map(id => {dataSources.requests.softDelete(id)})
```
Comment (meganniu) [inline server/graphql/resolvers.ts:10]: We might need more error handling for softDelete
My recommendation: if `requests.softDelete(id)` doesn't succeed, then stop immediately and return `ServerResponseInterface` with error

Same for `RequestGroup` soft  delete. 

Thoughts?

Comment (kevin-wangg) [reply server/graphql/resolvers.ts:10]: what does 'not succeed' mean? like if the id doesn't exist? or if the request is already deleted? or both

Comment (meganniu) [reply server/graphql/resolvers.ts:10]: if requests.softDelete(id) returns an error like
```
 {
          "success": false,
          "message": error._message,
          "id": id
        }
```

Comment (guy-ro) [reply server/graphql/resolvers.ts:10]: I think a problem is that we shouldn't necessarily return a ServerResponseInterface from the datasource. Instead, simply return something similar to the Promise for the query for mutations. Then in the resolvers, create a corresponding ServerResponseInterface at the end of the resolver. In the body of the resolver, make calls to the datasource create/update/etc. as needed and keep track if any errors happened.

This way, the resolver can successfully manage things like: resolver needs to do (1) and (2), but if (1) fails then don't do (2), and if (2) fails,. then revert (1).

**Comment by guy-ro** [comment]
I know that it isn't in the scope of this PR, but if it is suitable, could you add the corresponding work for requestGroups? It should be quick if using your helper function for requestTypes.

File: server/graphql/resolvers.ts (hunk 1 above)
Comment (meganniu) [reply server/graphql/resolvers.ts:10]: > Instead, simply return something similar to the Promise for the query for mutations. 

Sorry, not sure what "the Promise for the query for mutations" is referring to. clarification?

Comment (meganniu) [reply server/graphql/resolvers.ts:10]: moving this convo onto slack...

**Pull request merged**